### PR TITLE
sys/puf_sram: counter based seed after soft reset

### PR DESCRIPTION
**Pull request by PeterKietzmann**
### Contribution description

The SRAM seeder implements a detection mechanisms which avoids creating seeds from low entropy memory pattern after a soft reset. Currently, the old seed is reused in these cases which provides zero variance after a soft reset (see #12166). 

This PR adds a soft reset counter which is mixed into the "old" seed to create a "new" seed for the soft restart. The counter is set to zero after a power cycle was detected and a fresh seed is created from the SRAM pattern as usual.

Additionally, this PR contains a commit that disables the "interactive sync" module in the test application because there is no testrunner available anyway. Hitting "s" after each reboot is annoying.

### Testing procedure

1. I've added `printf("puf_sram_softreset_cnt: [0x%08" PRIX32 "]\n", puf_sram_softreset_cnt);` to the SRAM seeder [test application](https://github.com/RIOT-OS/RIOT/blob/master/tests/puf_sram/main.c#L28) and repeatedly reset the to watch the counter increase and reset after a power cycle. 

2. I've set up a nucleo-l073rz board similar to the [Example Setup](https://github.com/RIOT-OS/RIOT/tree/master/tests/puf_sram#example-setup) **but** applied continuous power and toggled the NRST pin. Then I ran [example_test.py](https://github.com/RIOT-OS/RIOT/blob/master/tests/puf_sram/tests/example_test.py) (500 iterations by default) which lead to a total min. entropy of ~55% between soft resets. This is not perfect and might improve with an other hash function, but it is a huge improvement to zero change.

### Issues/PRs references
 #12166, #12991

### CODE REVIEW

**Comment by fjmolinas** [comment]
> I'll test after the release is over.

Please ping me if I forget and @maribu doesn't test before that.

**Comment by maribu** [comment]
While this succeeds in generating a new seed every time, the first byte of the seed on the board I tested did not change at all between boots. But that is expected, as the DEK hash generates related outputs for related inputs

File: sys/puf_sram/puf_sram.c [hunk 1]
```diff
@@ -51,5 +56,12 @@ bool puf_sram_softreset(void)
         return 0;
     }
     puf_sram_state = 1;
+
+    /* increment number of detected soft resets */
+    puf_sram_softreset_cnt++;
+
+    /* generate alterntive seed value */
+    puf_sram_seed^=puf_sram_softreset_cnt;
```
Comment (maribu) [inline sys/puf_sram/puf_sram.c]:
```suggestion
    puf_sram_seed ^= puf_sram_softreset_cnt;
```

**Comment by PeterKietzmann** [comment]
@maribu thanks for testing. I've added your suggested change and squashed it immediately to run Murdock right now.

**Comment by PeterKietzmann** [comment]
Green lights btw

**Pull request merged**